src/allmydata/storage_client.py

```diff
@@ -1017,7 +1017,7 @@ def advise_corrupt_share(
             shnum,
             reason,
     ):
-        self._rref.callRemote(
+        return self._rref.callRemote(
```

Comment:
Ah... So while there are no callers using this return value in master, the tests newly added to this branch *do* use this return value...

So maybe while there are no current callers of the other changed methods, maybe there will be future callers.  And I guess that's what you meant by "may as well fix them all" (which I did not previously appreciate).

That makes more sense, maybe the other changes are fine.  I still have one concern, though, which is that with the change they fit the anti-pattern of library code logging and swallowing a failure and then giving the Deferred back to application code.

Maybe callers need to be responsible for this logging now?  I see how that's significantly more of a pain in the ass because there may be more than one caller and they may have no error handling logic...

But ... is this exactly the anti-pattern that `callRemoteOnly` encoded before we stopped using it?  Eh no, not *quite*.  I see that `callRemoteOnly` is:

```
    def callRemoteOnly(self, _name, *args, **kwargs):
        # the remote end will not send us a response. The only error cases
        # are arguments that don't match the schema, or broken invariants. In
        # particular, DeadReferenceError will be silently consumed.
        d = defer.maybeDeferred(self._callRemote, _name, True,
                                args, kwargs)
        del d
        return None
```

Instead, it relies on the garbage collector to trigger the logging (complete with the ritualistic `del d` to appease the gods of ... something)..

I guess that's a *worse* anti-pattern.  Before we had "completion notification unavailable, failure suppressed, failure reason non-deterministically logged".  Now we have "completion notification available, failure suppressed, failure reason deterministically logged early".

I guess I shouldn't complain about that.

Reply:
So I'll go with this "this is slight improvement so may as well merge as is" as my interpretation of the above :)